### PR TITLE
Do not remove an image if it is used by container(s)

### LINKED ISSUE #13: Return additional info on verbose image status call

### LINKED ISSUE #77: Return an error when removing image being used

### PR DESCRIPTION
**Pull request by sashayakovtseva**
A new interface - `image.Secretary` - is introduced to prevent images from being removed when they are used by containers. Secretary is implemented by `image.shelf` and used by `RuntimeService` which is responsible for calling `Borrow` and `Return`. `image.shelf` is also used to return additional info upon verbose image status call.

A bit of refactoring: slice util functions are moved to a separate package, not found errors in `index` package are united in a single one.

Fixes #77. 
Closes #13.

### CODE REVIEW

File: pkg/server/image/shelf.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+	"github.com/sylabs/cri/pkg/image"
+	"github.com/sylabs/cri/pkg/index"
+	"github.com/sylabs/cri/pkg/slice"
+)
+
+var errIsBorrowed = fmt.Errorf("image is used by at least one container")
+
+type shelf struct {
+	indx *index.ImageIndex
```
[the remark this answers is not in the document]
Comment (sashayakovtseva) [reply pkg/server/image/shelf.go]: This way it should be replaced all over the repo, perhaps a separate cleanup PR should be created

File: pkg/image/secretary.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// anyone who relies on image presence, e.g. container will 'borrow'
+// an image for its lifetime and 'return' at the end.
+// Secretary may be used to prevent image from removal.
+type Secretary interface {
+	// Find searches for image by id or reference. When image is
+	// not found ErrNotFound is returned.
+	Find(id string) (*Info, error)
+	// Borrow notifies that image is used by someone and should not be
+	// removed until Return with the same parameters is called.
+	Borrow(id, who string) error
```
[the remark this answers is not in the document]
Comment (sashayakovtseva) [reply pkg/image/secretary.go]: I see your point, updated comments

File: pkg/server/image/shelf.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bauerm97) [reply pkg/server/image/shelf.go]: OK, that's fair then. If we want to change `indx` -> `idx` let's do it in a separate PR and all at once!

**Comment by sashayakovtseva** [comment]
As @cclerget suggested, it might be better to do same thing in a different way. CRI could lock `/dev` directory to check loop devices when image is about to get removed.

**Comment by sashayakovtseva** [comment]
Closing in favor of #139

**Pull request closed without merging**